### ISSUE TITLE
expect(foo).toHaveMethod('bar'); should output what method name was missing.

### ISSUE THREAD

**Issue by JessicaSachs** [JamieMason/Jasmine-Matchers#40]
I'd like to be able to read what method name is missing from an object when an assertion fails. Thoughts?

```
var foo = {
  notbar: function () { return }
};

expect(foo).toHaveMethod('bar');

// Current: Expected Object({ notbar: Function }) to have method.
// Desired: Expected Object({ notbar: Function }) to have method 'bar'.
```

**Comment by JamieMason** [comment]
Thanks Jessica, I think this is a regression in the latest version. I'll pick it up as soon as I can.

**Comment by JessicaSachs** [comment]
Thanks for the quick response. I'll think I'll give it a shot and submit a PR :)

**Comment by JamieMason** [comment]
You're welcome. 

BTW if you do, bonus points for adding tests for failure message output. This issue has shown me that those are a blind spot at the moment. 

Much appreciated, cheers.

**Comment by JessicaSachs** [comment]
OK, i have a loose handle on the framework. :)

I think you'll have a quicker time getting through this than I will -- but I'd love to code review && maybe talk about adding tests for `memberMatcherFactory`, or however you decide to handle the integration between jasmine's `util.buildFailureMessage` and the matcher factory.

Findings:
This issue affects everything that asserts via `lib/factory/jasmine-v2/memberMatcherFactory`

I don't see anything written to explicitly handle passing arguments through to the failure message, so I'd need to start by understanding what `util.buildFailureMessage` actually does and how to pass through the matcher arguments.

**Comment by JessicaSachs** [comment]
(also I didn't test this in jasmine V1)

**Comment by JamieMason** [comment]
Hey, 
Some background info I can give you is that this used to be handled by [these private utils](https://github.com/JamieMason/Jasmine-Matchers/blob/db02b30000c717c588c2e029724dd356fd7bc203/src/wrapper/private.js#L107-L155) but during the [recent refactor](https://github.com/JamieMason/Jasmine-Matchers/commit/e02982ea40d85761f9d0a9eae8358512a6fad133) I think there have been some regressions. If you'd like to apply this to the new codebase, that would be a great help - otherwise I'll get to it as soon as I can.

Thanks again for your help, let me know if you need any more info.

**Comment by JamieMason** [comment]
As far as testing the failure output etc I'm not sure just yet as it's not something I've done before. I'll look around and see what I can find.

**Comment by JamieMason** [comment]
Released in 2.0.0-beta3.
